internal/workflow.go

```diff
@@ -46,92 +46,218 @@ var (
+		// (for signal channels) the data in the channel.
+		//
+		// If decoding or assigning fails:
+		// - an error will be logged
+		// - the value will be dropped from the channel
+		// - Receive will automatically try again
+		// - This will continue until a successful value is found, or the channel is emptied and it resumes blocking.
+		//   Closed channels with no values will always succeed, but they will not change valuePtr.
+		//
+		// Go would normally prevent this at compile time, but the same cannot be done here.
+		// If you need to "try" to assign to multiple things, similar to a Future you can use:
+		// - for signal channels, a []byte pointer.  This will give you the raw data that Cadence received, and no
+		//   decoding will be attempted, so you can try it yourself.
+		// - for other channels, an interface{} pointer.  All values are interfaces, so this will never fail, and you
+		//   can inspect the type with reflection or type assertions.
 		Receive(ctx Context, valuePtr interface{}) (more bool)
 
-		// ReceiveAsync try to receive from Channel without blocking. If there is data available from the Channel, it
-		// assign the data to valuePtr and returns true. Otherwise, it returns false immediately.
+		// ReceiveAsync try to receive from Channel without blocking.
+		// If there is data available from the Channel, it assigns the data to valuePtr and returns true.
+		// Otherwise, it returns false immediately.
+		//
+		// Decoding or assigning failures are handled like Receive.
 		ReceiveAsync(valuePtr interface{}) (ok bool)
 
-		// ReceiveAsyncWithMoreFlag is same as ReceiveAsync with extra return value more to indicate if there could be
+		// ReceiveAsyncWithMoreFlag is same as ReceiveAsync, with an extra return to indicate if there could be
 		// more value from the Channel. The more is false when Channel is closed.
+		//
+		// This behaves the same as
+		//  select {
+		//  case v, more := <- aChannel: ok = true
+		//  default: ok = false
+		//  }
+		//
+		// Decoding or assigning failures are handled like Receive.
 		ReceiveAsyncWithMoreFlag(valuePtr interface{}) (ok bool, more bool)
 
 		// Send blocks until the data is sent.
+		//
+		// This behaves the same as `aChannel <- v`
 		Send(ctx Context, v interface{})
 
-		// SendAsync try to send without blocking. It returns true if the data was sent, otherwise it returns false.
+		// SendAsync will try to send without blocking.
+		// It returns true if the data was sent (i.e. there was room in the buffer, or a reader was waiting to receive
+		// it), otherwise it returns false.
+		//
+		// This behaves the same as:
+		//  select {
+		//  case aChannel <- v: ok = true
+		//  default: ok = false
+		//  }
 		SendAsync(v interface{}) (ok bool)
 
-		// Close close the Channel, and prohibit subsequent sends.
+		// Close closes the Channel, and prohibits subsequent sends.
+		// As with a normal Go channel that has been closed, sending to a closed channel will panic.
```

Comment:
Closing a _signal_ channel in tests works... but sending to it causes a panic in the testing stack, which wouldn't happen in a real workflow.
So I haven't documented how closed signal channels behave yet, because I don't really know for sure.